src/plugins/console/public/application/containers/editor/monaco/monaco_editor.tsx

```diff
@@ -6,16 +6,105 @@
  * Side Public License, v 1.
  */
 
-import React, { FunctionComponent, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { CodeEditor } from '@kbn/code-editor';
 import { css } from '@emotion/react';
 import { CONSOLE_LANG_ID } from '@kbn/monaco';
-import { useEditorReadContext } from '../../../contexts';
+import { parse } from 'query-string';
+import { debounce } from 'lodash';
+import { decompressFromEncodedURIComponent } from 'lz-string';
+import { i18n } from '@kbn/i18n';
+import { useServicesContext, useEditorReadContext } from '../../../contexts';
+import { DEFAULT_INPUT_VALUE } from '../../../../../common/constants';
 
-export const MonacoEditor: FunctionComponent = () => {
+export interface EditorProps {
+  initialTextValue: string;
+}
+
+interface QueryParams {
+  load_from: string;
+}
+
+export const MonacoEditor = ({ initialTextValue }: EditorProps) => {
+  const {
+    services: { notifications },
+  } = useServicesContext();
   const { settings } = useEditorReadContext();
 
-  const [value, setValue] = useState('GET /.kibana/_search');
+  const [value, setValue] = useState(initialTextValue);
+
+  useEffect(() => {
+    const readQueryParams = () => {
+      const [, queryString] = (window.location.hash || window.location.search || '').split('?');
+
+      return parse(queryString || '', { sort: false }) as Required<QueryParams>;
+    };
+
+    const loadBufferFromRemote = (url: string) => {
+      // Normalize and encode the URL to avoid issues with spaces and other special characters.
+      const encodedUrl = new URL(url).toString();
+      if (/^https?:\/\//.test(encodedUrl)) {
+        const loadFrom: Record<string, any> = {
+          url,
+          // Having dataType here is required as it doesn't allow jQuery to `eval` content
+          // coming from the external source thereby preventing XSS attack.
+          dataType: 'text',
+          kbnXsrfToken: false,
+        };
+
+        if (/https?:\/\/api\.github\.com/.test(url)) {
```

Comment:
@ElenaStoeva I was not able to find any examples of loading a request from github, do you know how this works?

Reply:
I copied this logic from the original [code editor](https://github.com/elastic/kibana/blob/ffcc1ffa3381d07b397178846c04a6b1d5fcaba2/src/plugins/console/public/application/containers/editor/legacy/console_editor/editor.tsx#L129), but I'm also not sure how exactly this works. I tracked down the PR where this was added it seems it is some legacy functionality - see https://github.com/elastic/kibana/pull/52814/files#r357049056.

Reply:
okay, thanks a lot for the info, I think we can remove that since there is no documentation about loading from github